ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,240 @@
+  }
+
+  /**
+   * Gets the metadata of the account in {@link JSONObject}.
+   * @return The metadata of the account in {@link JSONObject}.
+   * @throws JSONException If fails to compose the metadata in {@link JSONObject}.
+   */
+  public JSONObject toJson() throws JSONException {
+    JSONObject metadata;
+    switch (version) {
+      case ACCOUNT_METADATA_VERSION_1:
+        metadata = new JSONObject();
+        metadata.put(ACCOUNT_METADATA_VERSION_KEY, version);
+        metadata.put(ACCOUNT_ID_KEY, id);
+        metadata.put(ACCOUNT_NAME_KEY, name);
+        metadata.put(ACCOUNT_STATUS_KEY, status);
+        JSONArray containerArray = new JSONArray();
+        for (Container container : containerIdToContainerMap.values()) {
+          containerArray.put(container.toJson());
+        }
+        metadata.put(CONTAINERS_KEY, containerArray);
+        break;
+
+      default:
+        throw new IllegalArgumentException("Unsupported account metadata version=" + version);
+    }
+    return metadata;
+  }
+
+  /**
+   * Gets the status of the account.
+   * @return The status of the account.
+   */
+  public AccountStatus getStatus() {
+    return status;
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container id.
+   * @param containerId The id of the container to get.
+   * @return The {@link Container} of this account with the specified id, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerById(short containerId) {
+    return containerIdToContainerMap.get(containerId);
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container name.
+   * @param containerName The name of the container to get.
+   * @return The {@link Container} of this account with the specified name, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByName(String containerName) {
+    return containerNameToContainerMap.get(containerName);
+  }
+
+  /**
+   * Gets all the containers of this account in a list.
+   * @return All the containers of this account.
+   */
+  public List<Container> getAllContainers() {
+    return Collections.unmodifiableList(new ArrayList<>(containerIdToContainerMap.values()));
+  }
+
+  /**
+   * Adds a {@link Container} to this account and updates internal maps accordingly.
+   * @param container The container to update this account.
+   */
+  private void updateContainerMap(Container container) {
+    containerIdToContainerMap.put(container.getId(), container);
+    containerNameToContainerMap.put(container.getName(), container);
+  }
+
+  /**
+   * Generates a {@link String} representation that uniquely identifies this account. The string
+   * is in the format of {@code Account[id]}.
+   * @return The {@link String} representation of this account.
+   */
+  @Override
+  public String toString() {
+    return "Account[" + getId() + "]";
```

Comment:
The reason is to make it to be consistent with `toString()` for container. In the container's version, the container instance does not have account name information, only account id.